### PR TITLE
RFC022 Use ERB as the default templating language in our Rails projects

### PR DESCRIPTION
**Pull request by mec**
This is something I have felt on almost every Rails project I've worked on - I am hoping I've kept personal views out of this and focused on what I beleive is the reason for the change - making our code more approachable to all.

Looking forward to hearing you thoughts.

Haml comes with a steep learning curve for a large number of folks and makes our
code less approachable. Erb ships with Rails and bares close similarity to many
other templating languages.

~~Don't default to using Haml in Rails projects, use erb instead~~

Use ERB as the default templating language in our Rails projects

### CODE REVIEW

**Comment by pezholio** [comment]
I'm 100% in agreement with this, but I think it should probably be rephrased as "Use ERB as our templating language", rather than expressly saying "Don't use this tool"

File: rfc-023-dont-default-to-haml.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Don't default to using Haml in Rails projects, use erb instead
```
[the remark this answers is not in the document]
Comment (mec) [reply rfc-023-dont-default-to-haml.md]: Oh hound, it's not a git commit message!?!

**Comment by robbiepaul** [comment]
Yeah I agree, I much prefer using ERB and it's easier to pick up. Although I've found it's led to more markup errors/mistakes in HTML than something like HAML but that's a separate issue.

**Comment by mec** [comment]
@pezholio I'd like to keep the haml in there as I do feel it is haml that make the code less approachable, but I agree the tone could change.

**Comment by mec** [comment]
@robbiepaul agreed! I was close to adding something about linting or checking but I wanted to keep the message here as clear as possible and feel like you, that is seperate issue.

**Comment by Floppy** [comment]
100% agree, but yeah, reorder it to be a positive choice for ERB, then say things like "other tools like Haml could be used if appropriate, but only if they deliver significant advantage to the team" or something.

**Comment by lozette** [comment]
100% agree. I ~hate~ find haml really hard to work with and would prefer to use erb. ~I also struggle with phrasing things in a positive way~

**Comment by erbridge** [comment]
Also, my revisions based on an early draft Meyric sent me earlier before I realized he'd already posted it 😊 . Use what you will :). I do think there's something about debugging that would be worth including.

```md
# Stop using Haml in Rails projects by default

## Summary

Haml comes with a steep learning curve for a large number of folks and makes our
code less approachable and harder to debug.

## Problem

Haml is powerful, but it's very different from the default markup language of
the web: HTML. Most templating languages (including ERB) extend HTML, making
them easier to pick up by people familiar with HTML, which most of the tech team
will have some familiarity with.

Haml is also abstracted from what renders in the browser, making it harder to
debug. With an HTML based templating language one can look at what rendered in
the browser and see direct correlations with the template that it rendered from.
With Haml, that markup needs to be translated, adding cognitive load to that
process.

## Proposal

An application SHOULD NOT use Haml as its default templating language.
Applications SHOULD use ERB as their templating language by default. An
application MAY use Haml if the team working on it feel its benefits outweigh
its drawbacks.

## Next steps

- Remove Haml from the
  [dxw rails-template](https://github.com/dxw/rails-template), switching back to
  ERB, the Rails default
```

File: rfc-023-dont-default-to-haml.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Use ERB as the default templating language in our Rails projects
+
+## Summary
+
+Haml comes with a steep learning curve for a large number of folks and makes our
+code less approachable and harder to debug. Erb ships with Rails and bares close
```
Comment (CristinaRO) [inline rfc-023-dont-default-to-haml.md]: Typo: "bares" should be "bears"

File: rfc-023-dont-default-to-haml.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+will have some familiarity with.
+
+Haml is also abstracted from what renders in the browser, making it harder to
+debug. With an HTML based templating language one can look at what rendered in
+the browser and see direct correlations with the template that it rendered from.
+
+Asking folks to learn an abstraction over something as familiar as HTML could be
+placing an unnecessary burden on their time and energies.
+
+As Rails ships with ERB as it's own default, large amounts of Rails
```
Comment (CristinaRO) [inline rfc-023-dont-default-to-haml.md]: Typo: "it's" should be "its"

File: rfc-023-dont-default-to-haml.md (hunk 2 above)
Comment (mec) [reply rfc-023-dont-default-to-haml.md]: Thank you!

**Comment by evilstreak** [comment]
For my own usage, I agree – I prefer using something closer to HTML.

Personal preference aside, I think the fact that ERB involves a lower barrier to contributions (due to being closer to HTML, which is table stakes for web development) is an important one for making it more accessible to new staff and easier to maintain for client teams we hand over to. It's very in line with our development principle: "Use simple, conventional technology".

I've approved because I'm happy with the decision. I would strongly encourage you to rewrite not just the title, but the whole RFC to be "use ERB" first, rather than "don't use HAML". If you feel it's important to mention HAML throughout because it's our current choice, then please do – but also consider that long in the future once ERB is the default and HAML is rarely used, this should stand as useful decision text by itself, without the context that we used to have a lot of HAML.

**Comment by mec** [comment]
Thank you everyone for the contributions - you all describe my thoughts in a more elegant way than I ever could hope to!

I'll have another go over the RFC copy and see what comes out…

**Comment by mec** [comment]
@jacksonj04 sorry for the massive delay on this, been on client days for so long! This has been completely re-written, could you take a look and let me know what you think?

Once I get your approval I can ship this!

File: rfc-022-use-erb-as-default-templating-language.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Use ERB as the default templating language in our Rails projects
```
[the remark this answers is not in the document]
Comment (erbridge) [reply rfc-022-use-erb-as-default-templating-language.md:1]: @mec Suggest `Use ERB by default in our Rails projects`

**Pull request merged**